chainer/functions/normalization/batch_normalization.py

```diff
@@ -727,11 +741,14 @@ def __init__(self, x, gamma, key_axis, inference=False):
         # into a 2-dim array with channels as second dim and m=<product
         # of all dimensions except the 2nd dimension> as the first
         # dimension.
+        # ideep only support float32
```

Comment:
Previously ideep was tested with other dtypes. I think it should work as long as `x.dtype` and `gamma.dtype` are the same.